wgengine/userspace.go

```diff
@@ -1076,6 +1078,23 @@ func (e *userspaceEngine) linkChange(changed bool, cur *interfaces.State) {
 	health.SetAnyInterfaceUp(up)
 	e.magicConn.SetNetworkUp(up)
 
+	// Hacky workaround for Linux DNS issue 2458: on
+	// suspend/resumse or whenever NetworkManager is started, it
+	// nukes all systemd-resolved configs. So reapply our DNS
+	// config on major link change.
+	if runtime.GOOS == "linux" && changed {
```

Comment:
Presumably we only need to do this for the systemd-resolved case. I guess it has the downside of rewriting /etc/resolv.conf more often in the Direct case, if users have written scripts to make local modifications after Tailscale comes up. However its also likely to fix other cases of broken DNS, just by rewriting more often.

Reply:
Elsewhere we have an issue suggesting we fsnotify resolv.conf and rewrite it on change, so we always win. I think that’s generally the right attitude: if we think we own it, own it.